### PR TITLE
Add Cast history back

### PR DESCRIPTION
**Pull request by houseroad**
Add the old version of Cast back.

### CODE REVIEW

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using namespace ONNX_NAMESPACE;
 
 ONNX_OPERATOR_SCHEMA(Cast)
+    .SinceVersion(6)
```
Comment (linkerzhang) [inline onnx/defs/tensor/defs.cc]: 6 or 7?

**Comment by linkerzhang** [comment]
@bddppq this is not related to this PR. I realized that "shape" was also not since version 1, right? :)

File: onnx/defs/tensor/defs.cc (hunk 1 above)
Comment (houseroad) [reply onnx/defs/tensor/defs.cc]: We introduced this change in opset 6, I think it's better to keep the truth in our code.

Comment (houseroad) [reply onnx/defs/tensor/defs.cc]: Any particular reason to use 7? If there is, we can use 7.

**Comment by bddppq** [comment]
@linkerzhang yeah that was not :(

File: onnx/defs/tensor/defs.cc (hunk 1 above)
Comment (bddppq) [reply onnx/defs/tensor/defs.cc]: This should be 6, which was the time we introduced this op.

**Pull request merged**